### ISSUE TITLE
[Bug]: Not able to Render Vectors with customized css- heavy size data (more than 10000 objects)

### ISSUE THREAD

**Issue by HarshaZeitview** [rnmapbox/maps#3489]
### Mapbox Implementation

Mapbox

### Mapbox Version

default

### React Native Version

0.73.1

### Platform

iOS, Android

### `@rnmapbox/maps` version

10.1.23

### Standalone component to reproduce

```javascript
import MapboxGL from '@rnmapbox/maps';
import React, {useRef, useMemo} from 'react';
import {StyleSheet} from 'react-native';

const BugReportExample = () => {

  // this is dynamic data which comes from the server. now just making it static, it contains around 10000 data in below format
  // contains uuid and the colour code with respect to it. the below pattern go's on upto the data we have.
  const drawPoints = ["4a5800cc-6343-4969-8d74-e40cc5c5f26a", "#FF0000", "5b6911dd-6343-4969-8d74-e40cc5c5f26a", "#AF0320"];

  const mapRef = useRef();
  const cameraRef = useRef();

  const styleUrl = "https://api.mapbox.com/styles/v1/mapbox/satellite-v9/static";

  const fillLayerStyle = useMemo(() => {
    return {
      fillColor: [
        'match',
        ['get', 'uuid'],
        ...drawPoints,
        'rgba(227, 220, 216, 0.8)',
      ],
    };
  }, [drawPoints]);

  return (
    <MapboxGL.MapView
      ref={mapRef}
      style={styles.root}
      logoEnabled={false}
      styleURL={styleUrl}
      zoomLevel={15}
      maxZoomLevel={20}
      minZoomLevel={15}
      compassEnabled={true}
      compassImage="compass"
      compassViewPosition={1}
      attributionEnabled={false}
      scaleBarEnabled={false}>
      <MapboxGL.Camera
        ref={cameraRef}
        animationMode={'flyTo'}
        animationDuration={1000}
        followUserMode={'normal'}
        zoomLevel={15}
        centerCoordinate={[-74.00597, 40.71427]}
      />
      <MapboxGL.VectorSource
        id="module-layer"
        tileUrlTemplates={["mvt URL"]} // this is mvt tile url
        hitbox={{
          width: 10,
          height: 20,
        }}>
        <MapboxGL.FillLayer
          id="module-layer-fill"
          sourceID="module-layer"
          sourceLayerID={"sourceLayerID"}
          style={fillLayerStyle}
        />
      </MapboxGL.VectorSource>
    </MapboxGL.MapView>
  );
}

const styles = StyleSheet.create({
  root: {
    flex: 1,
    backgroundColor: 'rgba(24, 26, 28, 0.4)',
    overflow: 'hidden',
  },
});

export default React.memo(BugReportExample);
```

### Observed behavior and steps to reproduce

So Now when we run the above code if the data points has lesser data then we can able to see the markings on the map, if the data is beyond 10000 data points then we can't be able to see markings on the map.

**Screenshot with around 500 data points**
<img width="324" alt="Screenshot 2024-05-13 at 8 34 19 PM" src="https://github.com/rnmapbox/maps/assets/136063212/f22c61d2-fdbf-4c8e-833c-2bc10e45f9a1">

**Screenshot with more than  10000 data points**
<img width="324" alt="Screenshot 2024-05-13 at 8 38 44 PM" src="https://github.com/rnmapbox/maps/assets/136063212/65d698cd-2f40-47fd-bada-fcf0a1fe66fa">

So as you see nothing was marking on it.
Earlier we use the **ShapeSource** with **FillLayer** but as it has larger Geojson data then app is immediately crashing.

### Expected behavior

Need to mark the points on the Map even the the data size is more then 1M. The Marking of points on the map should be immediately reflective.

**Comment by mfazekas** [comment]
@HarshaZeitview can you please change the example so it actually reproduces the issue? 

You can use something like
```
[...Array(10000)].map((_, i) => /* expression to generate I-th sample data */);
```

**Comment by HarshaZeitview** [comment]
Hi @mfazekas we're encountering with loading a large dataset of over 100,000 data points onto the map, we've observed that attempting to load the entire dataset causes the application to crash. Even when we attempt to load a subset of the data, such as more than 10,000 points, the rendering process fails. We've noticed that removing the drawPoints from the fillLayer expression allows the MVT tile to load successfully with default colors. However, when we attempt to add coloring based on the UUID, the rendering fails

So on the above code the size of the drawPoints is more than 10k, this list will come from the server and we will push this into the fillLayer expression.
